### ISSUE TITLE
Support for bi-directional mindmaps

### ISSUE THREAD

**Issue by jacktanner** [dundalek/markmap#3]
Does the parser assume that there's exactly one root node? It'd be nice if it would allow for a mindmap that has a root node at the center, and child nodes both left and right of the root. Here's a simple test case:

```markdown
## This is h2

## And another h2

# And an h1, ouch
```

**Comment by dundalek** [comment]
If you are missing a heading at certain level it creates a blank node to preserve hierarchy.
If you have multiple top-level nodes it creates extra root node to wrap them.

Your example will look like this:

![image](https://cloud.githubusercontent.com/assets/755611/12010054/b458e9a8-ac95-11e5-8f32-31bb8682d07d.png)

**Comment by jacktanner** [comment]
Exactly. Those two are useful heuristics, but perhaps we could relax them, and work towards nodes on left and right of root.

Where in the code are these two heuristics? It might help in the conversation.

**Comment by dundalek** [comment]
I think the bi-directional display is not only limited to your case but can be used generally even for proper trees. Therefore it is the matter of the rendering and not parsing.

This [d3 example](http://bl.ocks.org/mbostock/3184089) can be used as a start, the changes are needed in [view.mindmap.js](https://github.com/dundalek/markmap/blob/master/view.mindmap.js). I think the trickiest part is to come up with good heuristics how to divide nodes between two sides, so that it is reasonably balanced.

As for the two heuristics, they are located in [transform.headings.js](https://github.com/dundalek/markmap/blob/master/transform.headings.js).

**Comment by jacktanner** [comment]
Do you think it's reasonable to divide the nodes between the two sides NOT using a heuristic but instead using the order in the file? If an h2 comes before an h1, the node of the h2 goes left of the h1, otherwise it goes right.

**Comment by mralexgray** [comment]
I just woke up.... so this might make no sense.. but if "to the left" directives are implemented, it might be nice to also add implicit rules to force "to the top", and "to the bottom", too.. no?

**Comment by fzyzcjy** [comment]
Hi, is there any updates? Thanks
